sycl/source/detail/platform_impl.cpp

```diff
@@ -337,7 +337,7 @@ platform_impl::get_devices(info::device_type DeviceType) const {
   PlatformImplPtr PlatformImpl = getOrMakePlatformImpl(MPlatform, *MPlugin);
   std::transform(
       PiDevices.begin(), PiDevices.end(), std::back_inserter(Res),
-      [this, PlatformImpl](const RT::PiDevice &PiDevice) -> device {
+      [PlatformImpl](const RT::PiDevice &PiDevice) -> device {
```

Comment:
This is to correct this:
platform_impl.cpp: error: lambda capture 'this' is not used [-Werror,-Wunused-lambda-capture]